### ISSUE TITLE
Use standard weight decay for RetinaNet

### ISSUE THREAD

**Issue by LukeWood** [keras-team/keras-cv#1397]
currently we have a `weight_decay` argument in compile().  This deviates from the Keras standard, where weight decay is implemented via `regularization_loss` or by the optimizer.

We should try to remove this - and instead use the optimizer weight decay arguments.

One consideration is excluding `batchnorm` layers:
```
        with tf.GradientTape() as tape:
            total_loss = self.compute_loss(x, boxes, classes, training=True)

            reg_losses = []
            if self.weight_decay:
                for var in self.trainable_variables:
                    if "bn" not in var.name:
                        reg_losses.append(self.weight_decay * tf.nn.l2_loss(var))
                l2_loss = tf.math.add_n(reg_losses)
            total_loss += l2_loss
```

**Comment by ianstenbit** [comment]
We axed this from RetinaNet to instead favor the `weight_decay` arg in the new optimizers
